promising/promises.py

```
@@ -0,0 +1,549 @@
+from typing import Any, Generic, Optional
+from weakref import WeakSet
+
+from promising.configs import PromiseConfig
+from promising.errors import NoCurrentPromiseError, NoParentPromiseError
+from promising.sentinels import NOT_SET, Sentinel
+from promising.types import T_co
+
+_promise_name_counter = itertools.count(1)
+
+
+def get_current_promise(raise_if_none: bool = True) -> Optional["Promise[Any]"]:
```

Ruff (0.14.14)

[error] Boolean-typed positional argument in function definition

(FBT001)